lightning/src/routing/scorer.rs

```diff
@@ -205,19 +230,29 @@ impl Default for ScoringParameters {
 			base_penalty_msat: 500,
 			failure_penalty_msat: 1024 * 1000,
 			failure_penalty_half_life: Duration::from_secs(3600),
+			overuse_penalty_start_1024th: 1024 / 8,
+			overuse_penalty_msat_per_1024th: 20,
 		}
 	}
 }
 
 impl<T: Time> routing::Score for ScorerUsingTime<T> {
 	fn channel_penalty_msat(
-		&self, short_channel_id: u64, _send_amt_msat: u64, _chan_capacity_sat: u64, _source: &NodeId, _target: &NodeId
+		&self, short_channel_id: u64, send_amt_msat: u64, chan_capacity_sat: u64, _source: &NodeId, _target: &NodeId
 	) -> u64 {
 		let failure_penalty_msat = self.channel_failures
 			.get(&short_channel_id)
 			.map_or(0, |value| value.decayed_penalty_msat(self.params.failure_penalty_half_life));
 
-		self.params.base_penalty_msat + failure_penalty_msat
+		let send_1024ths = send_amt_msat.checked_mul(1024).unwrap_or(u64::max_value())
+			/ (chan_capacity_sat.checked_mul(1000).unwrap_or(1));
```

Comment:
Why not use `unwrap_or(u64::max_value())` here to indicate a large channel capacity?

Reply:
I'd think we'd want to default to penalizing the channel - the channel is obviously bogus if we can't multiply its sat value by 1000, so we shouldn't divide by much.

Reply:
Would it make sense to return an infinite penalty in that case?

```rust
let chan_capacity_msat = match chan_capacity_sat.checked_mul(1000) {
    Some(chan_capacity_msat) => chan_capacity_msat,
    None => return u64::max_value(),
}
```